pkg/network/tracer/testutil/testdns/test_dns_server.go

```diff
@@ -20,20 +21,34 @@ import (
 var globalServer *server
 var globalServerError error
 var serverOnce sync.Once
+var non53ServerOnce sync.Once
+
+const localhostAddr = "127.0.0.1"
+
+// GetServerIPPort53 is a helper that calls GetServerIP with port 53
+func GetServerIPPort53(t *testing.T) net.IP {
+	return GetServerIP(t, 53)
+}
 
 // GetServerIP returns the IP address of the test DNS server. The test DNS server returns canned responses for several
 // known domains that are used in integration tests.
 //
 // see server#start to see which domains are handled.
-func GetServerIP(t *testing.T) net.IP {
+func GetServerIP(t *testing.T, port int) net.IP {
+	if port != 53 {
+		non53ServerOnce.Do(func() {
+			globalServer.Start("tcp", port)
```

Comment:
How does this work if the dns server with port 53 is already running since you're using the same `globalServer` variable for both?

Reply:
The constructor `newServer()` used to take care of creating interfaces, and no longer does. We can dispense with storing the `server`  in a global variable completely now that the server uses loopback